src/perturbed/additive.jl

```diff
@@ -7,12 +7,19 @@ See also: [`AbstractPerturbed`](@ref).
 
 Reference: <https://arxiv.org/abs/2002.08676>
 """
-struct PerturbedAdditive{F,R<:AbstractRNG,S<:Union{Nothing,Int}} <: AbstractPerturbed
+struct PerturbedAdditive{F,R<:AbstractRNG,S<:Union{Nothing,Int},B} <: AbstractPerturbed{B}
     maximizer::F
     ε::Float64
     nb_samples::Int
     rng::R
     seed::S
+
+    function PerturbedAdditive{F,R,S,B}(
+        maximizer::F, ε::Float64, nb_samples::Int, rng::R, seed::S
+    ) where {F,R<:AbstractRNG,S<:Union{Nothing,Int},B}
+        @assert B isa Bool
```

Comment:
I was also wondering how to constrain the type of a "value" type parameter, if you have any clues I'll take them!

Reply:
This is the only solution I found: https://discourse.julialang.org/t/restrict-type-of-value-type-parameter/12127